src/types.rs

```diff
@@ -483,22 +483,16 @@ impl Rewrite for ast::Ty {
                 ty.rewrite(context, budget, offset + 1).map(|ty_str| format!("({})", ty_str))
             }
             ast::TyTup(ref tup_ret) => {
-                let inner = try_opt!(tup_ret.iter()
-                                            .map(|item| item.rewrite(context, width, offset))
-                                            .fold(Some("".to_owned()),
-                                                  |sum, x| {
-                                                      match (sum, x) {
-                                                          (Some(sum), Some(x)) => {
-                                                              if sum == "" {
-                                                                  // First item.
-                                                                  Some(x)
-                                                              } else {
-                                                                  Some(sum + ", " + &x)
-                                                              }
-                                                          }
-                                                          _ => None,
-                                                      }
-                                                  }));
+                let inner = if let [ref item] = &**tup_ret {
+                    try_opt!(item.rewrite(context, width, offset).map(|rewrite| rewrite + ","))
+                } else {
+                    let rewrites: Option<Vec<_>>;
+                    rewrites = tup_ret.iter()
```

Comment:
Honestly, I probably saw someone do it on reddit or IRC at some point...

...but it is indeed listed under implementors here -- https://doc.rust-lang.org/nightly/std/iter/trait.FromIterator.html. Note that it's not just `Option<Vec<T>>` but any `Option<A>` where `A` implements `FromIterator<T>`